File: Server/Server/Server/Model/Data/IMessageDAO.cs
using Server.Model.DTO;

namespace Server.Model.Data
{
    /// <summary>
    /// Classe permettant d'accéder aux données des messages.
    /// </summary>
    public interface IMessageDAO
    {
        /// <summary>
        /// Permet de récupérer la conversation entre deux utilisateurs.
        /// </summary>
        /// <param name="username1">le nom d'utilisateur du premier utilisateur</param>
        /// <param name="username2">le nom d'utilisateur du deuxième utilisateur</param>
        /// <returns>La liste des messages échangés triés par date</returns>
        public List<MessageDTO> GetConversation(string username1, string username2);
    }
}

File: Client/src/app/Model/DAO/IMessageDAO.ts
import { Observable } from 'rxjs';
import { MessageDTO } from '../DTO/MessageDTO';

/**
 * Interface pour le service MessageDAO, décrivant les méthodes disponibles.
 */
export interface IMessageDAO {
  /**
   * Récupère la conversation entre l'utilisateur courant et un autre utilisateur.
   * @param token Le token de l'utilisateur courant
   * @param usernameRecipient Le nom d'utilisateur du destinataire
   * @returns Un Observable contenant la liste des messages dans une propriété Messages.
   */
  GetConversation(token: string, usernameRecipient: string): Observable<{ Messages: MessageDTO[] }>;
}
